tests/test_wordpress_api.py

```diff
@@ -27,245 +20,181 @@ def raise_for_status(*args):
 
 
 class TestWordPressApi(unittest.TestCase):
-    @patch("canonicalwebteam.http.CachedSession.get")
-    def test_getting(self, get):
-
-        get.return_value = MockResponse()
-
-        article = api.get_article(slug="test")
-        get.assert_called_once_with(
-            "https://admin.insights.ubuntu.com/wp-json/wp/v2/"
-            + "posts?slug=test&tags=&tags_exclude="
-        )
-        self.assertEqual(article, "hello_test")
-
-    @patch("canonicalwebteam.http.CachedSession.get")
-    def test_excluding_article(self, get):
-
-        get.return_value = MockResponse()
-
-        article = api.get_article(slug="test", tags_exclude=[1234, 5678])
-        get.assert_called_once_with(
-            "https://admin.insights.ubuntu.com/"
-            + "wp-json/wp/v2/posts?slug=test&tags=&tags_exclude=1234,5678"
-        )
-        self.assertEqual(article, "hello_test")
-
-    @patch("canonicalwebteam.http.CachedSession.get")
-    def test_including_article(self, get):
-
-        get.return_value = MockResponse()
-
-        article = api.get_article(slug="test", tags=[1234, 5678])
-        get.assert_called_once_with(
-            "https://admin.insights.ubuntu.com/"
-            + "wp-json/wp/v2/posts?slug=test&tags=1234,5678"
-            + "&tags_exclude="
-        )
-        self.assertEqual(article, "hello_test")
-
-    @patch("canonicalwebteam.http.CachedSession.get")
-    def test_including_and_excluding_article(self, get):
-
-        get.return_value = MockResponse()
-
-        article = api.get_article(
-            slug="test", tags=[1234, 5678], tags_exclude=[9876]
+    @vcr.use_cassette("fixtures/vcr_cassettes/articles_with_metadata.yaml")
```

Comment:
IMO you are testing way too much stuff. Don't test the API but your code.

What I want to know here is:

- Have you made an API call(s) when calling the function?
- What is happening if API replies 40x/50x?
- What happens if JSON broken?
- If there is a minimum of logic what are the edge cases?

Reply:
Id rather keep the api test in as well, to make sure that regressions get caught.

For the error testing I have created https://github.com/canonical-web-and-design/canonicalwebteam.blog/issues/70

Reply:
This will be discussed together.